packages/storage-web-api/src/controllers/submit-website-scan-controller.ts

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import * as ApiContracts from 'api-contracts';
+import * as StorageDocuments from 'storage-documents';
+import { RestApiConfig, ServiceConfiguration } from 'common';
+import { inject, injectable } from 'inversify';
+import _ from 'lodash';
+import { ContextAwareLogger } from 'logger';
+import { ApiController, HttpResponse, WebApiErrorCodes, WebsiteProvider, WebsiteScanProvider } from 'service-library';
+import { createWebsiteScanApiResponse, WebsiteScanDocumentResponseConverter } from '../converters/website-scan-document-response-converter';
+import { SubmitWebsiteScanRequestValidator } from '../request-validators/submit-website-scan-request-validator';
+
+const defaultFrequencyKeys: { [key in StorageDocuments.ScanType]: keyof RestApiConfig } = {
+    a11y: 'defaultA11yScanFrequency',
+    privacy: undefined,
+    security: undefined,
+};
+
+@injectable()
+export class SubmitWebsiteScanController extends ApiController {
+    public readonly apiVersion = '1.0';
+
+    public readonly apiName = 'storage-web-api-submit-website-scan';
+
+    public constructor(
+        @inject(ServiceConfiguration) protected readonly serviceConfig: ServiceConfiguration,
+        @inject(ContextAwareLogger) logger: ContextAwareLogger,
+        @inject(SubmitWebsiteScanRequestValidator) requestValidator: SubmitWebsiteScanRequestValidator,
+        @inject(WebsiteProvider) private readonly websiteProvider: WebsiteProvider,
+        @inject(WebsiteScanProvider) private readonly websiteScanProvider: WebsiteScanProvider,
+        private readonly convertWebsiteScanDocumentToResponse: WebsiteScanDocumentResponseConverter = createWebsiteScanApiResponse,
+    ) {
+        super(logger, requestValidator);
+    }
+
+    public async handleRequest(): Promise<void> {
+        this.logger.setCommonProperties({ source: 'submitWebsiteScanRESTApi' });
+
+        const websiteScanRequest = this.tryGetPayload<ApiContracts.WebsiteScanRequest>();
+        this.logger.setCommonProperties({ websiteId: websiteScanRequest.websiteId });
+
+        let website: StorageDocuments.Website;
+        try {
+            website = await this.websiteProvider.readWebsite(websiteScanRequest.websiteId);
+        } catch (e) {
+            this.logger.logError('Unable to create websiteScan because website does not exist');
+            this.context.res = HttpResponse.getErrorResponse(WebApiErrorCodes.resourceNotFound);
```

Comment:
There is no exception parsing here to confirm that website is missing or we have network error for instance. In the  `WebsiteProvider.readWebsite()` func we can drop `client.ensureSuccessStatusCode(response)` call and add `throwIfNotSuccess=false` option to `cosmosContainerClient.readDocument()`. Then cosmos provider should return empty result with error code that can be validate here.